packages/financial-templates-lib/src/logger/Logger.ts

```diff
@@ -32,21 +32,35 @@ import { createTransports } from "./Transports";
 import { botIdentifyFormatter, errorStackTracerFormatter, bigNumberFormatter } from "./Formatters";
 import { delay } from "../helpers/delay";
 
-import type { Logger as _Logger } from "winston";
+import { Logger as _Logger } from "winston";
 import type * as Transport from "winston-transport";
 
-// This async function can be called by a bot if the log message is generated right before the process terminates.
-// This method will check if the AugmentedLogger's isFlushed is set to true. If not, it will block until such time
-// that it has been set to true. Note that each blocking transport should implement this isFlushed bool to prevent
-// the logger from closing before all logs have been propagated.
-export async function waitForLogger(logger: AugmentedLogger) {
-  while (!logger.isFlushed) await delay(0.5); // While the logger is not flushed, wait for it to be flushed.
+// Custom interface for transports that includes the isFlushed flag
+interface FlushableTransport extends Transport {
+  isFlushed: boolean;
 }
 
-export interface AugmentedLogger extends _Logger {
-  isFlushed: boolean;
+// Custom type guard function to check if a transport is of type FlushableTransport
+function isFlushableTransport(transport: Transport): transport is FlushableTransport {
+  return "isFlushed" in transport && typeof transport.isFlushed === "boolean";
+}
+
+// Function to check that all flushable transports attached to logger are in a flushed state.
+function isLoggerFlushed(logger: AugmentedLogger): boolean {
+  return logger.transports.filter(isFlushableTransport).every((transport) => transport.isFlushed);
 }
 
+// This async function can be called by a bot if the log message is generated right before the process terminates.
+// This method will check if all transports attached to AugmentedLogger having isFlushed property have it is set to
+// true. If not, it will block until such time that all these transports have been flushed. Note that each blocking
+// transport should correctly modify its isFlushed state to prevent the logger from closing before all logs have been
+// propagated.
+export async function waitForLogger(logger: AugmentedLogger): Promise<void> {
+  while (!isLoggerFlushed(logger)) await delay(0.5); // While the logger is not flushed, wait for it to be flushed.
+}
+
+export type AugmentedLogger = _Logger;
```

Comment:
Even if isFlushed is redundant on the logger, I keep AugmentedLogger interface since there is other outstanding PR that expects AugmentedLogger being extended _Logger interface